tests/test_enterprise/api/test_views.py

```diff
@@ -627,20 +614,175 @@ def test_enterprise_customer_courses(
-    """
-    Enterprise Catalog API endpoint validations.  These have been split into a separate class
-    due to an apparent test state conflict with the general API view validations above.
-    """
+    @ddt.data(
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_LIST_ENDPOINT,
+            False, False,
+            '',
+            {},
+            {'detail': "User must be a staff user or "
+                       "associated with the endpoint's appointed common Enterprise Customer."},
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_LIST_ENDPOINT,
+            False, True,
+            '',
+            {},
+            {'detail': "User must be a staff user or "
+                       "associated with the endpoint's appointed common Enterprise Customer."},
+        ),
+        (
+            ENTERPRISE_CUSTOMER_CATALOG_LIST_ENDPOINT,
+            True, True,
+            '',
+            {
+                'count': 3,
+                'next': 'http://testserver/enterprise/api/v1/enterprise_catalogs/?page=3&page_size=1',
+                'previous': 'http://testserver/enterprise/api/v1/enterprise_catalogs/?page_size=1',
+                'results': [fake_catalog_api.FAKE_SEARCH_ALL_COURSE_RESULT],
+            },
+            {
+                'count': 3,
+                'next': 'http://testserver/enterprise/api/v1/enterprise_catalogs/?page=3&page_size=1',
+                'previous': 'http://testserver/enterprise/api/v1/enterprise_catalogs/?page_size=1',
+                'results': [fake_catalog_api.FAKE_SEARCH_ALL_COURSE_RESULT],
+            },
+        ),
+    )
+    @ddt.unpack
+    def test_enterprise_customer_catalogs_list(self, url, has_linked_user, has_existing_catalog, querystring,
+                                               mocked_search_results, expected_catalogs):
+        """
+        Make sure the Enterprise Customer's Catalog view returns correctly paginated data.
+        """
+        enterprise_customer = factories.EnterpriseCustomerFactory(
+            uuid=fake_catalog_api.FAKE_UUIDS[0]
+        )
+        if has_linked_user:
+            factories.EnterpriseCustomerUserFactory(
+                user_id=self.user.id,
+                enterprise_customer=enterprise_customer
+            )
+            self.user.is_staff = True
+            self.user.save()
+        if has_existing_catalog:
+            factories.EnterpriseCustomerCatalogFactory(
+                uuid=fake_catalog_api.FAKE_UUIDS[-1],
+                enterprise_customer=enterprise_customer,
+                query=querystring
+            )
+        with mock.patch('enterprise.api.v1.views.CourseCatalogApiClient') as mock_catalog_api_client:
+            mock_catalog_api_client.return_value = mock.Mock(
+                get_paginated_search_results=mock.Mock(return_value=mocked_search_results),
+            )
+            response = self.client.get(url)
+            response = self.load_json(response.content)
+
```

Comment:
There are many pre-existing tests in this file using this style. I think it's a fine style -- separating the assertion(s) from the data gathering.